src/Pages/Signup/SignupInputForm/SignupInputForm.component.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from "react";
+
+import OverlapCheck from "./OverlapCheck/OverlapCheck.component";
+
+class SignupInputForm extends Component {
+  makeOverlapCheck = () => {
+    if (this.props.onOffCount === "1") {
+      return <OverlapCheck onCheckOverlap={this.props.onCheckId} />;
+    }
+    if (this.props.onOffCount === "2") {
+      return <OverlapCheck onCheckOverlap={this.props.onCheckEmail} />;
+    }
+  };
+
+  handleWriteData = (e) => {
+    this.props.onWriteData(e);
+  };
+
+  render() {
+    return (
+      <div className="SignupInputForm">
+        <div className="input-content">
+          {this.props.inputContent}
+          <span className="signup-ico">*</span>
+        </div>
+        <input
+          className="signup-input"
+          name={this.props.name}
+          placeholder={this.props.writeHolder}
+          type={this.props.textType}
+          onChange={this.handleWriteData}
+        ></input>
+        {this.makeOverlapCheck()}
```

Comment:
함수로 넣어주는 것보다 직접 jsx 요소가 표현되는 쪽이 가독성 측면에서 좋을 것 같습니다
```suggestion
        {onOffCount === "1" && <OverlapCheck onCheckOverlap={onCheckId} />}
        {onOffCount === "2" && <OverlapCheck onCheckOverlap={onCheckEmail} />}
```
이렇게 해주는건 어떨까 싶고, 한편으로는 어차피 Props로 받아온 값들이 들어오게 되는데 애초에 내려줄 때부터 결정해주는 것도 방법일 것 같습니다

Reply:
코드를 구현하는 자리가 짧으면 무조건 좋다고 생각했는데 감사합니다!